src/beamlime/logging/formatters.py

```diff
@@ -0,0 +1,298 @@
+    >>> header.format().replace(' ', '_')
+    TIME_____|_APPLICATION
+    >>> # 2
+    >>> message_column = LogColumn(variable_name='msg',\
+    >>> title='MESSAGE', style='{', ansi_colored=False)
+    >>> header = LogHeader(time_column, app_name_column, message_column)
+    >>> header.fmt
+    {asctime:8} | {ansi_color}{app_name}{reset_color} | {msg}
+    >>> # 3
+    >>> colored_message_column = LogColumn(variable_name='msg',\
+    >>> title='MESSAGE', style='{', ansi_colored=True)
+    >>> header = LogHeader(time_column, app_name_column, colored_message_column)
+    >>> header.fmt
+    {asctime:8} | {ansi_color}{app_name} | {msg}{reset_color}
+
+    """
+
+    @overload
+    def __init__(self, *fmt: str, padding: tuple, sep: _SepStyle):
+        ...
+
+    @overload
+    def __init__(self, *columns: LogColumn, padding: tuple, sep: _SepStyle):
+        ...
+
+    def __init__(
+        self,
+        *args: Union[str, LogColumn],
+        padding: tuple = (1, 1),
+        sep: _SepStyle = "|",
+    ):
+        self.padding = padding
+        self.sep = sep
+        if len(args) > 1 and isinstance(args[0], LogColumn):
+            self._init_from_columns(*args)
+
+        elif len(args) > 1 and isinstance(args[0], str):
+            self._init_from_literal_fmt(*args)
+
+    def _init_from_columns(self, *columns: LogColumn) -> None:
+        if len(columns) == 1:
+            self._fmt = columns[0].formatter
+        else:
+            sep = " " * self.padding[0] + self.sep + " " * self.padding[1]
+            _ansi_color_col = LogColumn("ansi_color", title="", ansi_colored=False)
+            _reset_color_col = LogColumn("reset_color", title="", ansi_colored=False)
+
+            _padded_columns = [_ansi_color_col] + list(columns) + [_reset_color_col]
+
+            # Color starting/terminating flags
+            _color_starts = [
+                True if column.ansi_colored and not p_column.ansi_colored else False
+                for column, p_column in zip(columns, _padded_columns[:-2])
+            ]
+            _color_ends = [
+                True if column.ansi_colored and not n_column.ansi_colored else False
+                for column, n_column in zip(columns, _padded_columns[2:])
+            ]
+
+            # Wrap columns with {ansi_color or color_key} and {reset_color}
+            _color_started_fmts = [
+                _ansi_color_col.formatter + column.formatter
+                if _color_starts[icol]
+                else column.formatter
+                for icol, column in enumerate(columns)
+            ]
+            _colored_fmts = [
+                fmt + _reset_color_col.formatter if _color_ends[icol] else fmt
+                for icol, fmt in enumerate(_color_started_fmts)
+            ]
+
+            self._fmt = sep.join(_colored_fmts)
+
+        for column in columns:
+            self.__setitem__(column.variable_name, column)
+
+    def _init_from_literal_fmt(self, *fmts: str) -> None:
```

Comment:
Instead of the custom parsing logic here, you can maybe use https://docs.python.org/3/library/string.html?highlight=string#string.Formatter to make this more robust.

Reply:
I need to update this part anyways. I'll do it when I work on this: #37